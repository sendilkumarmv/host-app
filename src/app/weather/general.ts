import { InjectionToken } from '@angular/core';

export const API_BASE_URL = new InjectionToken<string>('Base Url');



export function covertToDate(unix_timestamp, onlyTime = false) {
  var date = new Date(unix_timestamp * 1000);
  var hours = date.getHours();
  var minutes = "0" + date.getMinutes();
  var seconds = "0" + date.getSeconds();
  if (onlyTime)
    return hours + ':' + minutes.substr(-2) + ':' + seconds.substr(-2);
  else
    return date;
}

export function getFlags(countryCode: string) {
  return `https://www.countryflags.io/${countryCode.toLowerCase()}/flat/24.png`;
}
