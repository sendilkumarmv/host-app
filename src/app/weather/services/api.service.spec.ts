import { TestBed, fakeAsync,async, tick } from '@angular/core/testing';
import { LocationService } from './geolocation.service';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { ApiService } from './api.service';
import { API_BASE_URL } from '../general';
import { LocationCoordinates } from '../models/Coordinates.model';
import { of } from 'rxjs';


describe('Api Service ', () => {
  let locationService: LocationService;
  let apiService: ApiService;
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientTestingModule],
      providers: [
        ApiService,
        LocationService,
        { provide: API_BASE_URL, useValue: 'http://localhost:59377' }
      ]
    }).compileComponents();
    locationService = TestBed.get(LocationService);
    apiService = TestBed.get(ApiService);
  }));

  it('should return coordinates of the current location', async((done) => {
    spyOn(locationService, 'getCurrentCoordinates').and.returnValue(of({ lat: '12', lon: '77' }));
    const result$ = apiService.getCurrentLocation();
    let result: LocationCoordinates;
    result$.subscribe({
      next:(r) =>{
        result = r;
      }
    });
    // tick(60005);
    expect(result.lat).toMatchSnapshot('12');
    expect(result.lon).toMatchSnapshot('77');
  }));

})
