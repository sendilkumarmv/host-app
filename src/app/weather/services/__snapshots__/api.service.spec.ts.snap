// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api Service  should return coordinates of the current location: 12 1`] = `"12"`;

exports[`Api Service  should return coordinates of the current location: 77 1`] = `"77"`;
